client/asset/doge/doge.go

```diff
@@ -189,3 +215,44 @@ func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network)
 
 	return btc.BTCCloneWallet(cloneCFG)
 }
+
+// externalFeeEstimator gets the fee rate from the external API and returns it
+// in sats/vByte.
+func externalFeeEstimator(ctx context.Context, net dex.Network) (uint64, error) {
+	var url string
+	if net == dex.Testnet {
+		url = testnetExternalApiUrl
+	} else {
+		url = externalApiUrl
+	}
+	url = url + "fee/1"
+	ctx, cancel := context.WithTimeout(ctx, 4*time.Second)
+	defer cancel()
+	r, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return 0, err
+	}
+	httpResponse, err := http.DefaultClient.Do(r)
+	if err != nil {
+		return 0, err
+	}
+	var resp map[string]float64
+	reader := io.LimitReader(httpResponse.Body, 1<<20)
+	err = json.NewDecoder(reader).Decode(&resp)
+	if err != nil {
+		return 0, err
+	}
+	httpResponse.Body.Close()
+
+	feeBtcByte, ok := resp["feerate"]
```

Comment:
Right now the testnet URL is returning `{"feerate":-1,"blocks":25}`.

Need the sanity check, i.e. run it through the `dexdoge.DefaultFeeRateLimit` check that we do back in `FeeEstimator`